app/helpers/application_helper/button/basic.rb

```diff
@@ -64,6 +65,9 @@ def skipped?
 
   # Tells whether the button should displayed in the toolbar or not
   def visible?
+    # hide select power management buttons for Openstack::InfraManager
+    return false if %w(host_standby host_shutdown host_reboot host_reset).include?(@child_id) &&
+      @record.kind_of?(ManageIQ::Providers::Openstack::InfraManager)
```

Comment:
so  ApplicationHelper::Button::GenericFeatureButton already does that, so we just need the right 'supports' defined